server/model/monitor.js

```diff
@@ -645,6 +645,10 @@ class Monitor extends BeanModel {
                             retries = 0;
                             log.debug("monitor", `[${this.name}] timeout = ${timeout}`);
                             this.heartbeatInterval = setTimeout(safeBeat, timeout);
+
+                            bean.status = UP;
+                            this.prometheus?.update(bean, tlsInfo);
+
                             return;
```

Comment:
I am unsure if this return statement is correct in the first place.
This shortcuts a lot of logic, for example
https://github.com/louislam/uptime-kuma/blame/0060e46b91b108c6960858e12da5c2f312437a1b/server/model/monitor.js#L979

CC @kaysond you added this in https://github.com/louislam/uptime-kuma/pull/1428

I dont quite get why `No need to insert successful heartbeat for push type, so end here`. Could you explain this?

Reply:
I didn't actually add it, just moved it. See: https://github.com/louislam/uptime-kuma/pull/1428/files

But I think it's there because for push type monitors, the API call to the push URL adds the heartbeat to the DB. There's no need for the monitor to add the heartbeat, so everything else is skipped on purpose

Reply:
In that case maybe we should update the Prometheus metrics in the push route handler instead.